includes/class-suppression.php

```diff
@@ -84,60 +220,70 @@ function( $placement ) {
 	}
 
 	/**
-	 * Get whether ads should be displayed on a screen.
+	 * Internal method for determining if the page should display ads.
 	 *
-	 * @param int $post_id Post ID to check (optional, default: current post).
+	 * @param int $post_id Optional post ID to check for.
 	 *
 	 * @return bool
 	 */
-	public static function should_show_ads( $post_id = null ) {
-		$should_show = true;
-
-		if ( is_404() ) {
-			$should_show = false;
-		}
-
-		if ( is_singular() ) {
-			if ( null === $post_id ) {
-				$post_id = get_the_ID();
+	private static function internal_should_show_ads( $post_id = null ) {
+		$config = self::get_config();
+		if ( \is_singular() || ! empty( $post_id ) ) {
+			if ( ! $post_id ) {
+				$post_id = \get_the_ID();
 			}
-
-			if ( get_post_meta( $post_id, 'newspack_ads_suppress_ads', true ) ) {
-				$should_show = false;
+			if ( \get_post_meta( $post_id, 'newspack_ads_suppress_ads', true ) ) {
+				return false;
 			}
-		}
-
-		$global_suppression_config = GAM_Model::get_suppression_config();
-		if ( true === $global_suppression_config['tag_archive_pages'] ) {
-			if ( is_tag() ) {
-				$should_show = false;
+			if ( ! empty( $config['post_types'] ) && in_array( \get_post_type( $post_id ), $config['post_types'] ) ) {
+				return false;
 			}
-		} elseif ( ! empty( $global_suppression_config['specific_tag_archive_pages'] ) ) {
-			$suppressed_tags = $global_suppression_config['specific_tag_archive_pages'];
-			foreach ( $suppressed_tags as $tag_id ) {
-				if ( is_tag( $tag_id ) ) {
-					$should_show = false;
-				}
+			if ( ! empty( $config['tags'] ) && \has_tag( $config['tags'], $post_id ) ) {
+				return false;
+			}
+			if ( ! empty( $config['categories'] ) && \has_category( $config['categories'], $post_id ) ) {
+				return false;
 			}
 		}
-
-		if ( true === $global_suppression_config['category_archive_pages'] ) {
-			if ( is_category() ) {
-				$should_show = false;
+		if ( \is_archive() ) {
```

Comment:
Testing this with the post type suppression, I find myself wondering why suppressing ads on a post type doesn't also suppress ads for that post type's archive pages, like with categories and tags. Also wonder why there isn't a global option to suppress ads on all post type archives, like with categories and tags, too.

Reply:
Oops! I added that but accidentally removed it while cleaning things up.

Fixed in c7b94b4703969bb84836eb74d353bec6b8eff814

Reply:
Confirmed!